src/app/(nextjs_migration)/(guest)/breach-details/[breachName]/page.tsx

```diff
@@ -0,0 +1,368 @@
+import BreachDetailScanImage from "../../../../../client/images/breach-detail-scan.svg";
+import "../../../../../client/css/partials/breachDetail.css";
+import { getL10n } from "../../../../functions/server/l10n";
+import { getBreachByName } from "../../../../../utils/hibp";
+import {
+  getAllPriorityDataClasses,
+  getAllGenericRecommendations,
+} from "../../../../../utils/recommendations";
+import { BreachLogo } from "../../../../components/server/BreachLogo";
+import {
+  getBreachIcons,
+  getBreaches,
+} from "../../../../functions/server/getBreaches";
+import { Breach } from "../../../../transitionTypes";
+import { getLocale } from "../../../../functions/server/l10n";
+
+import Glyph1 from "../../../../../client/images/social-security-numbers.svg";
+import Glyph3 from "../../../../../client/images/passwords.svg";
+import Glyph4 from "../../../../../client/images/bank-account-numbers.svg";
+import Glyph5 from "../../../../../client/images/credit-cards.svg";
+import Glyph6 from "../../../../../client/images/credit-card-cvvs.svg";
+import Glyph7 from "../../../../../client/images/partial-credit-card-data.svg";
+import Glyph8 from "../../../../../client/images/ip-addresses.svg";
+import Glyph9 from "../../../../../client/images/historical-passwords.svg";
+import Glyph10 from "../../../../../client/images/security-questions-and-answers.svg";
+import Glyph11 from "../../../../../client/images/phone-numbers.svg";
+import Glyph12 from "../../../../../client/images/email-addresses.svg";
+import Glyph13 from "../../../../../client/images/dates-of-birth.svg";
+import Glyph14 from "../../../../../client/images/pins.svg";
+import Glyph15 from "../../../../../client/images/physical-addresses.svg";
+import MoreIcon from "../../../../../client/images/more.svg";
+
+const glyphs: Record<string, StaticImageData> = {
+  "social-security-numbers": Glyph1,
+  passwords: Glyph3,
+  "bank-account-numbers": Glyph4,
+  "credit-cards": Glyph5,
+  "credit-card-cvvs": Glyph6,
+  "partial-credit-card-data": Glyph7,
+  "ip-addresses": Glyph8,
+  "historical-passwords": Glyph9,
+  "security-questions-and-answers": Glyph10,
+  "phone-numbers": Glyph11,
+  "email-addresses": Glyph12,
+  "dates-of-birth": Glyph13,
+  pins: Glyph14,
+  "physical-addresses": Glyph15,
+};
+
+export default async function BreachDetail(props: {
+  params: { breachName: string };
+}) {
+  const l10n = getL10n();
+  const breachName = props.params.breachName;
+  const allBreaches = await getBreaches();
+  const breach = getBreachByName(allBreaches, breachName);
+  const breachLogos = await getBreachIcons(allBreaches);
+
+  return (
+    <div data-partial="breachDetail">
+      <header className="breach-detail-header">
+        <div className="breach-detail-meta">
+          <h1>
+            <BreachLogo breach={breach} logos={breachLogos} />
+            {breach.Title}
+          </h1>
+          {getBreachCategory(breach) === "website-breach" ? (
```

Comment:
Note (non-blocking and non-actionable for now): I know the task right now is only porting the current pages we have. For the next iteration, it would be great if we would define utilize enums more and minimize string comparisons.

Reply:
I think I might actually push back a little on enums: I believe the TS wouldn't have added those today, since they're a language-level feature added to JS, rather than just being annotations - and IIRC they're even incompatible with the enum proposal that's on the TC39 standardisation track.

That said, I do agree in spirit: I just think that string comparisons are fine, but would also like to tighten the type to a [union of string literals](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types).

Reply:
@Vinnl To confirm your intent, it would only return one of three possible values? 

`"data-aggregator-breach" | "website-breach" | "sensitive-breach"`

Reply:
I scraped each breach (because of course I would) and used `cheerio` to extract `.breach-detail-meta-more-info` on each breach detail's page into a `Set()` and it says we only have 3 unique breach types:
```js
Set(3) {
  'Website Breach',
  'Data Aggregator Breach',
  'Sensitive Website Breach'
}
```

Now that we're using TypeScript, I'd probably suggest moving those into an enum since it looks like they are hardcoded in a few places:
```sh
git grep -En "['\"](data-aggregator|website|sensitive)-breach['\"]" -- ':!*.ftl' | cat

src/views/partials/breachDetail.js:11:    return 'data-aggregator-breach'
src/views/partials/breachDetail.js:14:    return 'sensitive-breach'
src/views/partials/breachDetail.js:17:    return 'website-breach'
src/views/partials/breachDetail.js:19:  return 'data-aggregator-breach'
src/views/partials/breachDetail.js:103:  if (categoryId === 'data-aggregator-breach') {
src/views/partials/breachDetail.js:109:  } else if (categoryId === 'sensitive-breach') {
src/views/partials/breachDetail.js:138:      ${getBreachCategory(data.breach) === 'website-breach'
```

Reply:
@maxxcrawford Yep exactly, and TS would flip us off if we tried to use any other!